linkerd/client-policy/src/lib.rs

```diff
@@ -0,0 +1,146 @@
+#![deny(rust_2018_idioms, clippy::disallowed_methods, clippy::disallowed_types)]
+#![forbid(unsafe_code)]
+
+pub mod http;
+
+#[cfg(feature = "proto")]
+pub mod proto;
+
+use linkerd_addr::{Addr, NameAddr};
+// pub use linkerd_policy_core::{meta, Meta};
+use linkerd_proxy_api_resolve as resolve;
+use std::{fmt, net::SocketAddr, str::FromStr, sync::Arc};
+use tokio::sync::watch;
+
+type Backends = Arc<[Backend]>;
+
+#[derive(Clone, Debug)]
+pub struct Receiver {
+    inner: watch::Receiver<Policy>,
+}
+
+// TODO(eliza): this is looking a lot like the `Profile` type...could be made
+// generic eventually...
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Policy {
+    pub addr: Option<LogicalAddr>,
+    pub http_routes: Arc<[http::Route]>,
+    pub opaque_protocol: bool,
```

Comment:
How does this fit into the policy API in its current form? Where does this configuration come from?

Reply:
there's a proto module converting this...